readconf.c

```diff
@@ -1734,18 +1738,140 @@ read_config_file_depth(const char *filename, struct passwd *pw,
 
 	if ((f = fopen(filename, "r")) == NULL)
 		return 0;
+	if (flags & SSHCONF_CHECKPERM) {		
+#if WINDOWS
```

Comment:
Let's keep it this way for now. POSIX library should only contain POSIX compat calls (or their near equivalents).
Ideal solution for the above is to abstract out permission checks logic and have them implemented for Unix and Windows in their own source files.